tests/test_repository_lib.py

```diff
@@ -493,7 +501,107 @@ def test_generate_snapshot_metadata(self):
 
 
 
-  def test_generate_timestamp_metadata(self):
+  def test_generate_snapshot_metadata_without_length(self):
+    metadata_directory, version, expiration_date, targets_filename, \
+      storage_backend = self._setup_generate_snapshot_metadata_test()
+
+    snapshot_metadata = \
+      repo_lib.generate_snapshot_metadata(metadata_directory, version,
+                                          expiration_date,
+                                          targets_filename,
+                                          storage_backend,
+                                          consistent_snapshot=False,
+                                          use_length=False)
+    self.assertTrue(tuf.formats.SNAPSHOT_SCHEMA.matches(snapshot_metadata))
+
+    metadata_files_info_dict = snapshot_metadata['meta']
+    for metadata_filename in sorted(os.listdir(metadata_directory), reverse=True):
+
+      # In the metadata_directory, there are files with format:
+      # 1.root.json. The prefix number should be removed.
+      if metadata_filename[0].isdigit():
+        version_number, metadata_filename = metadata_filename.split('.', 1)
+
+      if metadata_filename.endswith('.json'):
+        if metadata_filename not in \
+            ['root.json', 'targets.json', 'timestamp.json', 'snapshot.json']:
+
+          # Check that length is not calculated but hashes is
+          self.assertTrue(
+            metadata_files_info_dict[metadata_filename].get('length') is None)
```

Comment:
This would be a bit cleaner if we used the `assertIn` method:
```suggestion
          self.assertIn('length', metadata_files_info_dict[metadata_filename])
```